MoltenVK/MoltenVK/GPUObjects/MVKSync.h

```diff
@@ -636,3 +636,56 @@ class MVKMetalCompiler : public MVKBaseObject {
 	std::string _compilerType = "Unknown";
 	MVKPerformanceTracker* _pPerformanceTracker = nullptr;
 };
+
+#pragma mark -
+#pragma mark MVKDeferredOperation
+
+/** Defines the function pointer for each dependent function*/
+union MVKDeferredOperationFunctionPointer
+{
+    // Empty until deferred functions from other extensions have been defined
+    // Planning to use std::functions
+};
+
+/** Manages what function is being deferred*/
+enum MVKDeferredOperationFunctionType
+{
+    // Empty until deferred functions from other extensions have been defined
+};
+
+class MVKDeferredOperation : public MVKVulkanAPIDeviceObject {
+public:
+    /** Returns the Vulkan type of this object. */
+    VkObjectType getVkObjectType() override { return VK_OBJECT_TYPE_DEFERRED_OPERATION_KHR; }
+    
+    /** Returns the debug report object type of this object. */
+    VkDebugReportObjectTypeEXT getVkDebugReportObjectType() override { return VK_DEBUG_REPORT_OBJECT_TYPE_UNKNOWN_EXT; }
+    
+    /** Joins the deferred operation */
+    VkResult join();
+    
+    /** Gets the max number of threads that can execute the deferred operation concurrently*/
+    uint32_t getMaxConcurrency() { return 1; } // Perhaps the number of CPU cores
```

Comment:
I expect this would be up to the particular deferred operation. For example, I suspect a "build acceleration structures" operation would only be able (under Metal, anyway) to use one thread per acceleration structure.